### ISSUE TITLE
make: -ldflags improvements 

### ISSUE THREAD

**Issue by mythi** [cri-o/cri-o#2979]
**Description**

1. `-X importpath.name=value` ldflags cannot be easily amended
2. `-X main.buildInfo=${SOURCE_DATE_EPOCH}` seems non-existent

I wanted to override `criocli.DefaultsPath` with make but it's not easy. A simple workaround would be to override `SHRINKFLAGS` with the default values plus my `-X ...DefaultsPath=` setting but not very nice.

While working with -X flags, I also noticed `main.buildInfo` is gone:

```
$ go tool nm bin/crio | grep -e buildInfo -e gitCommit
34adf70 d k8s.io/component-base/metrics.buildInfo
34363f0 d k8s.io/component-base/version.gitCommit
3436460 d main.gitCommit
2239460 r main.gitCommit.str
```

**Comment by saschagrunert** [comment]
Hey @mythi, I'm working on a fix for this in #3320 and #3321

**Comment by mythi** [comment]
@saschagrunert thanks!
